### PR TITLE
feat: Support ESLint 7.x

### LINKED ISSUE #318: eslint v7 support
We need a semver-minor to add eslint v7 to the peer deps.

### PR DESCRIPTION
**Pull request by MichaelDeBoey**
ESLint v7.0.0 is [released](https://eslint.org/blog/2020/05/eslint-v7.0.0-released) 🎉

---

Closes #318
Unblocks https://github.com/airbnb/javascript/issues/2223
Unblocks https://github.com/kentcdodds/eslint-config-kentcdodds/pull/64

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,9 +3,9 @@ language: node_js
 notifications:
   email: false
 node_js:
-  - '12'
-  - '10'
-  - '8'
+  - 14
+  - 12
+  - 10
```
Comment (ljharb) [inline .travis.yml:8]:
```suggestion
  - '14'
  - '12'
  - '10'
```

**Comment by ljharb** [comment]
Looks like the linter's failing

**Comment by MichaelDeBoey** [comment]
@ljharb Should I delete the 'node' run? 🤔
It's the same as the '14' run

**Comment by ljharb** [comment]
@MichaelDeBoey no, it's not the same; it's got the `LINTONLY` env var set, so that runs the linter and *not* the tests - none of the other jobs run the linter.

**Comment by MichaelDeBoey** [comment]
@ljharb It seems like tests are already failing on `master` (see #320):
https://travis-ci.org/github/sarbbottam/eslint-find-rules/builds/684856897

**Comment by MichaelDeBoey** [comment]
@ljharb All green 🙂

**Pull request merged**